pkl-core/src/main/java/org/pkl/core/ast/type/ResolveDeclaredTypeNode.java

```diff
@@ -73,11 +74,36 @@ protected VmTyped getImport(
     var result = module.getCachedValue(importName);
     if (result == null) {
       result = callNode.call(member.getCallTarget(), module, module, importName);
+
+      var importedModule = (VmTyped) result;
+      if (importedModule.isNotInitialized()
+          && importedModule.isModuleObject()
+          && importedModule.getModuleInfo().isAmend()) {
+        // this is an amending module. Try to find the prototype
+        var prototypeModule = findPrototypeModule(importedModule);
+        if (prototypeModule != null) {
+          result = prototypeModule;
+        }
+      }
+
       module.setCachedValue(importName, result);
     }
     return (VmTyped) result;
   }
 
+  private @Nullable VmTyped findPrototypeModule(VmTyped amendingModule) {
+    try {
+      var moduleInfo = amendingModule.getModuleInfo();
+      var amendedModuleKey = moduleInfo.getAmendedModuleKey();
+
+      if (amendedModuleKey != null) {
+        return VmLanguage.get(null).loadModule(amendedModuleKey);
```

Comment:
The amended module key itself may be amending another module; we need to keep going until we find the first non-amending module.

Also, pass the node in when looking up the reference

```suggestion
        return VmLanguage.get(this).loadModule(amendedModuleKey, this);
```